src/ArmTemplates/Extractor/EntityExtractors/PolicyExtractor.cs

```diff
@@ -18,43 +20,64 @@ public class PolicyExtractor : TemplateGeneratorBase, IPolicyExtractor
         public const string PoliciesDirectoryName = "policies";
         public const string GlobalServicePolicyFileName = "globalServicePolicy.xml";
 
-        readonly IPolicyApiClient policyApiClient;
+        public static string ProductPolicyFileNameFormat = "{0}-productPolicy.xml";
 
-        public PolicyExtractor(IPolicyApiClient policyApiClient)
+        readonly ILogger<PolicyExtractor> logger;
+        readonly IPolicyClient policyClient;
+
+        public PolicyExtractor(
+            ILogger<PolicyExtractor> logger,
+            IPolicyClient policyApiClient)
         {
-            this.policyApiClient = policyApiClient;
+            this.logger = logger;
+            this.policyClient = policyApiClient;
         }
 
-        public async Task<Template> GenerateGlobalServicePolicyTemplateAsync(ExtractorParameters extractorParameters, string baseFilesDirectory)
+        public async Task<PolicyTemplateResource> GenerateProductPolicyTemplateAsync(
+            ExtractorParameters extractorParameters, 
+            string productName,
+            string[] productResourceId,
+            string baseFilesGenerationDirectory)
         {
-            // extract global service policy in both full and single api extraction cases
-            Template armTemplate = this.GenerateEmptyPropertyTemplateWithParameters();
-            if (extractorParameters.PolicyXMLBaseUrl != null && extractorParameters.PolicyXMLSasToken != null)
+            var productPolicy = await this.policyClient.GetPolicyLinkedToProductAsync(extractorParameters, productName);
+
+            if (productPolicy is null)
             {
-                TemplateParameterProperties policyTemplateSasTokenParameterProperties = new TemplateParameterProperties()
-                {
-                    type = "string"
-                };
-                armTemplate.Parameters.Add(ParameterNames.PolicyXMLSasToken, policyTemplateSasTokenParameterProperties);
+                this.logger.LogWarning($"Policy not found for product '{productName}'");
+                return productPolicy;
             }
 
-            if (extractorParameters.PolicyXMLBaseUrl != null)
+            this.logger.LogDebug($"Policy linked to {productName} product found successfuly");
```

Comment:
Why debug? Maybe warning?

Reply:
It is not a very important message to log. Previously all the logs were written to *System.Console* output, and I wanted to make it more clear with levels of logs - I will try to **LogInfo** some general messages like `Starting execution of blablabla` and will try to **LogDebug** more internal stuff like `Found product with name 'my-super-product' in Azure`.

So that it is very clear in the console output what is happening and users can easily find all messages important for them